### PR TITLE
test: fix vscode jest timeout

### LINKED ISSUE #11710: Fetching gitlab submodules now fails with 403. oauth2 is stripped from URL.
### How are you running Renovate?

Self-hosted

### Please select which platform you are using if self-hosting.

_No response_

### If you're self-hosting Renovate, tell us what version of Renovate you run.

27.5.0

### Describe the bug

Since this PR was merged: https://github.com/renovatebot/renovate/pull/11615, fetching submodule fails with 403 errors on gitlab.

The debug logs looks ok, but right after, renovate is asking for a password showing the token in cleartext.
```
DEBUG: Found hostRules token for url https://oauth2:**redacted**@gitlab.example.com/myrepo.git
Password for 'https://Jy-5rWxLGyPRisfs9ZNt@gitlab.example.com': 
```

Reverting #11615 fix this issue.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

### PR DESCRIPTION
**Pull request by uncycler**
This only changes jest timeout for vscode

---

## Changes:

- Copy user attribute to token when the token is already part of it.
- Fix vscode jest arguments for remote container.

## Context:

The PR #11722 I proposed for fixing #11710 is not working. I'm not sure if it's the good way to do it, but this PR should definitely fix #11710. I've tried with bitbucket, github and gitlab.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
i think #11760 is a better solution

**Comment by uncycler** [comment]
Awesome, I missed it. Thanks!

**Comment by viceice** [comment]
Can you reopen this PR and only make the vscode jest setting please?

File: lib/util/git/url.spec.ts [hunk 1]
```diff
@@ -39,6 +39,13 @@ describe('util/git/url', () => {
       );
     });
 
+    it('returns https url with token with included username', () => {
+      hostRules.find.mockReturnValueOnce({ token: 'token' });
+      expect(
+        getRemoteUrlWithToken('https://oauth2:token@foo.bar/bar/foo')
+      ).toBe('https://oauth2:token@foo.bar/bar/foo');
+    });
+
```
Comment (viceice) [inline lib/util/git/url.spec.ts:48]:
```suggestion
```

File: lib/util/git/url.ts [hunk 2]
```diff
@@ -5,7 +5,11 @@ import * as hostRules from '../host-rules';
 export function getHttpUrl(url: string, token?: string): string {
   const parsedUrl = GitUrlParse(url);
 
-  parsedUrl.token = token;
+  if (parsedUrl.user.includes(token)) {
+    parsedUrl.token = parsedUrl.user;
+  } else {
+    parsedUrl.token = token;
+  }
```
Comment (viceice) [inline lib/util/git/url.ts:12]:
```suggestion
  parsedUrl.token = token;
```

**Review by viceice — APPROVED** [review]

**Pull request merged**